### PR TITLE
INTMDB-871: [Terraform] Improve acceptance test setup to run in parallel & against cloud-dev - backup

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR moves the backup tests from the old workflow ([~ 2 hours to run](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5280701123)) to a new one running the tests concurrently and against cloud-dev ([~ 23 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5309899434?pr=1250)). The new workflow splits the backup tests into multiple GitHub Jobs.

**GitHub Action Now**: [5309899434](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5309899434?pr=1250)
Screenshot 2023-06-19 at 11 26 06

**GitHub Action Before**:  [5280701123](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/5280701123)
Screenshot 2023-06-19 at 11 26 59

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: .github/workflows/acceptance-tests.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 name: 'Acceptance Tests'
 
 on:
-  workflow_dispatch: {}
-  workflow_call: 
+  workflow_dispatch: {} # workflow can be run manually
+  schedule:
+    - cron: "0 4 * * *" # workflow runs every day at 04:00 AM
+  workflow_call: # workflow runs after code-health
```
Comment (andreaangiolillo) [inline .github/workflows/acceptance-tests.yml:7]: Updated the workflow to also run all the tests against cloud-dev once per day at 4 AM UTC

File: .github/workflows/automated-test-acceptances.yml [hunk 2]
```diff
@@ -55,36 +55,6 @@ jobs:
           TEST_REGEX: "^TestAccAdv"
         run: make testacc
 
-  acceptance-tests-backup:
```
Comment (andreaangiolillo) [inline .github/workflows/automated-test-acceptances.yml:58]: removed the tests from the old workflow

File: mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule_test.go [hunk 3]
```diff
@@ -1,150 +0,0 @@
...
-	"os"
-	"testing"
-
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
-	"github.com/mwielbut/pointy"
-	matlas "go.mongodb.org/atlas/mongodbatlas"
-)
-
-func TestAccBackupDSCloudBackupSchedule_basic(t *testing.T) {
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule_test.go:14]: I have moved the data source tests into the resource tests. By doing this, there is no need to create new clusters specifically for these tests. Instead, we can efficiently reuse the clusters that are already created for the resource tests.

**Review by maastha — APPROVED** [review]
LGTM

**Review by martinstibbe — APPROVED** [review]
LGTM

**Pull request merged**